templates/scale_up_resources.j2

```diff
@@ -14,17 +14,57 @@ rsc_defaults \
 op_defaults \
     timeout="600"
 
-#
-# production HANA
-#
-
-{% set nic = "nic="~pillar.cluster.interface|default('eth0')|json if cloud_provider == "google-cloud-platform" else "nic="~pillar.cluster.interface|json if pillar.cluster.interface is defined else "" %}
+{%- if cloud_provider not in ["amazon-web-services"] %}
+# Create virtual ip. gcp must be changed when gcp-vpc-move-route RA is available
+{%- set nic = "nic="~pillar.cluster.interface|default('eth0')|json if cloud_provider == "google-cloud-platform" else "nic="~pillar.cluster.interface|json if pillar.cluster.interface is defined else "" %}
 
 primitive rsc_ip_{{ sid }}_HDB{{ instance }} ocf:heartbeat:IPaddr2 \
     params \
     ip="{{ data.virtual_ip }}" cidr_netmask="{{ 32 if cloud_provider == "google-cloud-platform" else data.virtual_ip_mask }}" {{ nic }} \
     op start timeout="20" op stop timeout="20" \
     op monitor interval="10" timeout="20"
+{%- endif %}
+
+# Platform dependant (stonith, virtual ip address, cib options, etc) resource
+{%- if cloud_provider == "amazon-web-services" %}
+
+property $id="cib-bootstrap-options" \
+    stonith-enabled="true" \
+    stonith-action="off" \
+    stonith-timeout="150s"
+
+primitive res_aws_stonith_{{ sid }}_HDB{{ instance }} stonith:external/ec2 \
+    params tag={{ data.instance_tag }} profile={{ data.cluster_profile}} \
+    op start interval=0 timeout=180 \
+    op stop interval=0 timeout=180 \
+    op monitor interval=120 timeout=60 \
+    meta target-role=Started
+
+primitive rsc_aws_vip_{{ sid }}_HDB{{ instance }} ocf:suse:aws-vpc-move-ip \
+    params ip={{ data.virtual_ip }} routing_table={{ data.route_table}} \
+    interface={{ pillar.cluster.interface|default('eth0')|json }} profile={{ data.cluster_profile}} \
+    op start interval=0 timeout=180 \
+    op stop interval=0 timeout=180 \
+    op monitor interval=60 timeout=60
+
+colocation col_saphana_ip_{{ sid }}_HDB{{ instance }} 2000: rsc_aws_vip_{{ sid }}_HDB{{ instance }}:Started msl_SAPHana_{{ sid }}_HDB{{ instance }}:Master
+
+{%- elif cloud_provider == "microsoft-azure" %}
+
+primitive rsc_socat_{{ sid }}_HDB{{ instance }} anything \
+    params binfile="/usr/bin/socat" cmdline_options="-U TCP-LISTEN:625{{ instance }},backlog=10,fork,reuseaddr /dev/null" \
+    op monitor timeout="20" interval="10" depth="0"
+
+group g_ip_{{ sid }}_HDB{{ instance }} rsc_ip_{{ sid }}_HDB{{ instance }} rsc_socat_{{ sid }}_HDB{{ instance }}
+
+colocation col_saphana_ip_{{ sid }}_HDB{{ instance }} 2000: g_ip_{{ sid }}_HDB{{ instance }}:Started msl_SAPHana_{{ sid }}_HDB{{ instance }}:Master
```

Comment:
Do we need to group if there is only 1 resource, here rsc_ip

Reply:
There are 2 resources: `rsc_ip` and `rsc_socat_`